addons/web/static/src/views/calendar/calendar_model.js

```diff
@@ -818,7 +818,7 @@ export class CalendarModel extends Model {
             label: isUserOrPartner
                 ? this.env._t("Everybody's calendars")
                 : this.env._t("Everything"),
-            active: previousAllFilter ? previousAllFilter.active : false,
+            active: previousAllFilter ? previousAllFilter.active : this.meta.allFilterLocalState,
```

Comment:
We can have x number of sections so using a simple boolean will set the filter in every section.